core/tests/puppeteer-acceptance-tests/user-utilities/curriculum-admin-utils.ts

```diff
@@ -0,0 +1,412 @@
+    await this.clickOn(createTopicButton);
+    await this.page.bringToFront();
+  }
+
+  async openTopicEditor(): Promise<void> {
+    await this.page.bringToFront();
+    await this.navigateToTopicAndSkillsDashboardPage();
+    await this.clickOn(topicsTab);
+    await this.clickOn(topic);
+  }
+
+  /**
+   * Function for creating a subtopic as a curriculum admin.
+   */
+  async createSubTopic(): Promise<void> {
+    await this.openTopicEditor();
+    await this.clickOn(addSubTopicButton);
+    await this.type(subTopicTitleField, 'Test Subtopic 1');
+    await this.type(subTopicUrlFragmentField, 'test-subtopic-one');
+    await this.clickOn(subTopicDescriptionEditorToggle);
+    await this.page.waitForTimeout(500);
+    await this.type(
+      richTextAreaField,
+      'This subtopic is to test curriculum admin utility.'
+    );
+    await this.clickOn(subtopicPhotoBoxButton);
+    await this.uploadFile(curriculumAdminThumbnailImage);
+    await this.page.waitForSelector(`${uploadPhotoButton}:not([disabled])`);
+    await this.clickOn(uploadPhotoButton);
+    await this.page.waitForSelector(`${createSubTopicButton}:not([disabled])`);
+    await this.page.waitForTimeout(500);
+    await this.clickOn(createSubTopicButton);
+    await this.page.waitForSelector(`${saveTopicButton}:not([disabled])`);
+    await this.page.waitForTimeout(500);
+    await this.clickOn(saveTopicButton);
+    await this.clickOn(closeSaveModalButton);
+  }
+
+  /**
+   * Function for creating a story for a certain topic.
+   */
+  async createStory(): Promise<void> {
+    await this.openTopicEditor();
+    await this.clickOn(addStoryButton);
+    await this.type(storyTitleField, 'Test Story 1');
+    await this.type(storyUrlFragmentField, 'test-story-one');
+    await this.type(
+      storyDescriptionField,
+      'This story is to test curriculum admin utility.'
+    );
+    await this.clickOn(storyPhotoBoxButton);
+    await this.uploadFile(curriculumAdminThumbnailImage);
+    await this.page.waitForSelector(`${uploadPhotoButton}:not([disabled])`);
+    await this.clickOn(uploadPhotoButton);
+    await this.page.waitForSelector(`${createStoryButton}:not([disabled])`);
+    await this.page.waitForTimeout(500);
+    await this.clickOn(createStoryButton);
+  }
+
+  /**
+   * Function for creating a chapter for a certain story.
+   */
+  async createChapter(explorationId: string): Promise<void> {
+    await this.clickOn(addChapterButton);
+    await this.type(chapterTitleField, 'Test Chapter 1');
+    await this.type(chapterExplorationIdField, explorationId);
+    await this.clickOn(chapterPhotoBoxButton);
+    await this.uploadFile(curriculumAdminThumbnailImage);
+    await this.page.waitForSelector(`${uploadPhotoButton}:not([disabled])`);
+    await this.clickOn(uploadPhotoButton);
+    await this.page.waitForSelector(`${createChapterButton}:not([disabled])`);
+    await this.page.waitForTimeout(500);
```

Comment:
Do you mean incorporating the wait in the base user click functions and such? That might be unwanted as they are common for all utilities

Although we try to avoid timeouts, these ones are very crucial for particular buttons that deal with creating, saving or publishing changes. 

The event we are waiting for is the buttons to be enabled or clickable but without the timeout, the "click" literally never happens for some reason - either it happens way too fast as the button enables or due to transitions, at least on my device. I've only left the ones in that are absolutely necessary.

Reply:
Can we then include the timeout in "wait for enabled/clickable", with a comment/explanation and a pointer to a specific example?

For specific cases with transitions I think it's fine to keep it here though, but in each such "exception case" there should be a comment pointing to the code that specifies the transition duration. Otherwise no one will know why it's 500 as opposed to 1000, 250, etc. and this will make it confusing to fix flakes.

Reply:
Sounds good, it's a recurring pattern and after a bit more investigation, seems to always happen when the "click" should happen after a modal closes or opens.

Will make a "wait for clickable" function with the appropriate explanation, thanks.